library/core/tests/nonzero.rs

```diff
@@ -328,3 +328,57 @@ fn test_nonzero_uint_rem() {
     let x: u32 = 42u32 % nz;
     assert_eq!(x, 2u32);
 }
+
+#[test]
+fn test_nonzero_checked_add() {
+    let one = NonZeroU32::new(1).unwrap();
+    let two = NonZeroU32::new(2).unwrap();
+    let max = NonZeroU32::new(u32::MAX).unwrap();
+    assert_eq!(Some(two), one.checked_add(1));
+    assert_eq!(None, max.checked_add(1));
+}
```

Comment:
The tests can be added as doc tests using the doc = concat behavior, see [here](https://github.com/rust-lang/rust/blob/795a934b51cb481ea3cb1cc8c3835a043a9e0102/library/core/src/num/int_macros.rs#L581) for example

Reply:
Okay, I'm doing this as well. Do you mean they should be *moved* there or *copied* there?

BTW I couldn't get `x test library/std` to run these tests, and had to fix them with the help of @rust-log-analyzer only, is it the wrong command to check that they pass at least on my machine?

Reply:
I'm thinking about moved, no need to test twice. In general for public items I prefer doc tests, though they are slower

This is in core so use library/core or just core